beluga/test/beluga/actions/test_overlay.cpp

```diff
@@ -0,0 +1,135 @@
+// Copyright 2024 Ekumen, Inc.
```

Comment:
It would be good to have a test that creates an action with a temporary mask, and uses it in a separate line:

```cpp
TEST(OverlayAction, Temporary) {
  auto input = std::vector{...};

  auto action = beluga::actions::overlay(std::vector{...}, true);

  input |= action;
  ASSERT_THAT(input, ::testing::Pointwise(::testing::Eq(), mask));
}
```